fedimint-client/src/lib.rs

```diff
@@ -1442,14 +1439,9 @@ impl ClientBuilder {
     }
 
     pub async fn load_decodable_client_secret<T: Decodable>(&mut self) -> anyhow::Result<T> {
-        let fake_notifications = Default::default();
         let mut dbtx = match self.db.as_ref().ok_or(anyhow!("No database provided"))? {
-            DatabaseSource::Fresh(db) => DatabaseTransaction::new(
-                db.begin_transaction().await,
-                Default::default(),
-                &fake_notifications,
-            ),
-            DatabaseSource::Reuse(db) => db.db().begin_transaction().await,
+            DatabaseSource::Fresh(db) => db.begin_transaction().await,
+            DatabaseSource::Reuse(client) => client.db().begin_transaction().await,
```

Comment:
Unrelated note: We likely want to remove this again. It was a workaround (451c97296bd0556f9516d5ffe84bd681cb1b02ac) for not being able to open the same DB again for recovery.